cpp/examples/abm.cpp

```diff
@@ -246,19 +246,35 @@ int main()
 
     //Set global infection parameters (similar to infection parameters in SECIR model) and initialize the world
     epi::GlobalInfectionParameters abm_params;
-    abm_params.set<epi::IncubationPeriod>({{epi::AbmAgeGroup::Count}, 4.});
-    abm_params.set<epi::SusceptibleToExposedByCarrier>({{epi::AbmAgeGroup::Count}, 0.02});
-    abm_params.set<epi::SusceptibleToExposedByInfected>({{epi::AbmAgeGroup::Count}, 0.02});
-    abm_params.set<epi::CarrierToInfected>({{epi::AbmAgeGroup::Count}, 0.15});
-    abm_params.set<epi::CarrierToRecovered>({{epi::AbmAgeGroup::Count}, 0.15});
-    abm_params.set<epi::InfectedToRecovered>({{epi::AbmAgeGroup::Count}, 0.2});
-    abm_params.set<epi::InfectedToSevere>({{epi::AbmAgeGroup::Count}, 0.03});
-    abm_params.set<epi::SevereToRecovered>({{epi::AbmAgeGroup::Count}, 0.1});
-    abm_params.set<epi::SevereToCritical>({{epi::AbmAgeGroup::Count}, 0.1});
-    abm_params.set<epi::CriticalToRecovered>({{epi::AbmAgeGroup::Count}, 0.02});
-    abm_params.set<epi::CriticalToDead>({{epi::AbmAgeGroup::Count}, 0.06});
-    abm_params.set<epi::RecoveredToSusceptible>({{epi::AbmAgeGroup::Count}, 0.});
+    
+    abm_params.set<epi::IncubationPeriod>({{epi::AbmAgeGroup::Count,epi::VaccinationState::Count}, 4.});
+    abm_params.set<epi::SusceptibleToExposedByCarrier>({{epi::AbmAgeGroup::Count,epi::VaccinationState::Count}, 0.02});
+    abm_params.set<epi::SusceptibleToExposedByInfected>({{epi::AbmAgeGroup::Count,epi::VaccinationState::Count}, 0.02});
+    abm_params.set<epi::CarrierToInfected>({{epi::AbmAgeGroup::Count,epi::VaccinationState::Count}, 0.15});
+    abm_params.set<epi::CarrierToRecovered>({{epi::AbmAgeGroup::Count,epi::VaccinationState::Count}, 0.15});
+    abm_params.set<epi::InfectedToRecovered>({{epi::AbmAgeGroup::Count,epi::VaccinationState::Count}, 0.2});
+    abm_params.set<epi::InfectedToSevere>({{epi::AbmAgeGroup::Count,epi::VaccinationState::Count}, 0.03});
+    abm_params.set<epi::SevereToRecovered>({{epi::AbmAgeGroup::Count,epi::VaccinationState::Count}, 0.1});
+    abm_params.set<epi::SevereToCritical>({{epi::AbmAgeGroup::Count,epi::VaccinationState::Count}, 0.1});
+    abm_params.set<epi::CriticalToRecovered>({{epi::AbmAgeGroup::Count,epi::VaccinationState::Count}, 0.02});
+    abm_params.set<epi::CriticalToDead>({{epi::AbmAgeGroup::Count,epi::VaccinationState::Count}, 0.06});
+    abm_params.set<epi::RecoveredToSusceptible>({{epi::AbmAgeGroup::Count,epi::VaccinationState::Count}, 0.1});
 
+    // Set each parameter for vaccinated people
+    for (int i = 0; i < (int)epi::AbmAgeGroup::Count; i++) {
```

Comment:
you can use age group to iterate instead of int to make the loop body clearer.

something like `for (auto age = epi::Index<AgeGroup>(0); age < epi::AbmAgeGroup::Count; ++age)` should work, 
then use `age` in the loop body directly instead of `(AbmAgeGroup)i`.

Reply:
Changed it to the for loop head which you suggested